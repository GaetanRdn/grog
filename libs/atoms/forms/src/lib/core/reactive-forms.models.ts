export type OnChangeFn<ValueType> = (value: ValueType | null) => void;
export type OnTouchedFn = () => void;
