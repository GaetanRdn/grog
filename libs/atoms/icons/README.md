# atoms-icons

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test atoms-icons` to execute the unit tests.
