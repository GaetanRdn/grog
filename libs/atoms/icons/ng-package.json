{
  "$schema": "../../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../../dist/libs/atoms/icons",
  "lib": {
    "entryFile": "src/index.ts"
  }
}
