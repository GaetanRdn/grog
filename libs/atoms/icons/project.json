{
  "projectType": "library",
  "root": "libs/atoms/icons",
  "sourceRoot": "libs/atoms/icons/src",
  "prefix": "grorg",
  "targets": {
    "build": {
      "executor": "@nrwl/angular:ng-packagr-lite",
      "outputs": ["dist/libs/atoms/icons"],
      "options": {
        "project": "libs/atoms/icons/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/atoms/icons/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/atoms/icons/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/atoms/icons"],
      "options": {
        "jestConfig": "libs/atoms/icons/jest.config.js",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/atoms/icons/src/**/*.ts", "libs/atoms/icons/src/**/*.html"]
      }
    }
  },
  "tags": []
}
