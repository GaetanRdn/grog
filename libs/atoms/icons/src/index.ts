export * from './lib/material-icon.component';
