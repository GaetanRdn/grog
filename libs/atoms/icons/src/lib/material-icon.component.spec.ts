import { Component } from '@angular/core';
import { TemplateLookup } from '@grorg/tests';
import { TestBed, waitForAsync } from '@angular/core/testing';
import { MaterialIconComponent } from '@grorg/atoms/icons';

describe('MaterialIconComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(
    waitForAsync(() => {
      TestBed.configureTestingModule({
        declarations: [MaterialIconComponent, HostComponent],
      }).compileComponents();
    })
  );

  beforeEach(() => {
    templateLookup = new TemplateLookup(HostComponent);
  });

  test.each(['home', 'verified'])('should create %s icon', (icon: string) => {
    templateLookup.hostComponent.content = icon;
    templateLookup.detectChanges();
    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });
});

@Component({
  template: ` <gro-material-icon>{{ content }}</gro-material-icon>`,
})
export class HostComponent {
  public content!: string;
}
