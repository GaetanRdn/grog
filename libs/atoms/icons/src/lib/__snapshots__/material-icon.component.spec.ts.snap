// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialIconComponent should create home icon 1`] = `
<gro-material-icon
  class="material-icons"
>
  home
</gro-material-icon>
`;

exports[`MaterialIconComponent should create verified icon 1`] = `
<gro-material-icon
  class="material-icons"
>
  verified
</gro-material-icon>
`;
